### ISSUE TITLE
[REQ] Allow Maven Plugin to explicitly specify method name

### ISSUE THREAD

**Issue by Emdee89** [OpenAPITools/openapi-generator#2761]
### Is your feature request related to a problem? Please describe.
The problem is that we have one Open API definition (i.e., one YAML file) describing two operations with the same name, each residing under a different URL hierarchy.

Examples:
- myEndpoint/myOperationContext1/myOperation
- myEndpoint/myOperationContext2/myOperation

Note that I have a different tag for the two operation contexts, i.e., I end up with two Java classes.

As I want to enforce meaningful Java method names, I define an operationId for each of them. As the operation is semantically doing the same thing, but in a different contexts, I would like to have the two methods to have the same name. Java-wise this is fine, as the two methods reside in different classes. However, Open API requires the operationId to be globally unique (I guess globally refers to "per file").

As a consequence, I cannot specify two operationIds with the same name and cannot name my methods properly.

## Describe the solution you'd like
I

**Comment by Emdee89** [comment]
Update: I found a kind of nice workaround for the described issue.
Instead of adding arbitrary chars that are stripped out during generation to the operationId, like `.`, you can add a prefix to the operationId to reflect the operation context. And then you can invoke the Maven Plugin with the parameter `removeOperationIdPrefix` set to `true`. Works quite nicely.

Still I think that a general mechanism to map operationIds to method names is desirable.

However, I can live with my workaround for now.

**Comment by edouardmercier** [comment]
I agree with @Emdee89, a global configuration parameter which would achieve this work-around by default, would be nice: when you want your generated methods to have the same name as their default "operationId" counterpart (which is the method name for instance in Java), having to add a prefix on all colliding web services operations is a bit tedious.
